### PR TITLE
Modify issues found by shellcheck and snyk

### PR DESCRIPTION
**Pull request by tenderbulk**

### CODE REVIEW

File: build/ubuntu-22.04/intel-mvp-tdx-guest-grub2/build.sh [hunk 1]
```diff
@@ -15,12 +15,12 @@ fi
 
 get_source() {
     echo "Get downstream source code..."
-    cd "${CURR_DIR}"
+    cd "${CURR_DIR}" || exit
```
Comment (kenplusplus) [inline build/ubuntu-22.04/intel-mvp-tdx-guest-grub2/build.sh:18]: why the operation of 'cd "$CURR_DIR"' could be fail?

File: build/ubuntu-22.04/intel-mvp-tdx-guest-grub2/build.sh [hunk 2]
```diff
@@ -15,12 +15,12 @@ fi
...
     if [[ ! -d ${PACKAGE}-${UPSTREAM_VERSION} ]]; then
 	git config --global --add safe.directory /github/workspace
         git clone --branch ${DOWNSTREAM_TAG} --single-branch --depth 1 \
             ${DOWNSTREAM_GIT_URI} ${PACKAGE}-${UPSTREAM_VERSION}
-        cd "${PACKAGE}-${UPSTREAM_VERSION}"
+        cd "${PACKAGE}-${UPSTREAM_VERSION}" || exit
```
Comment (kenplusplus) [inline build/ubuntu-22.04/intel-mvp-tdx-guest-grub2/build.sh:23]: Why need exit here?

File: build/ubuntu-22.04/intel-mvp-tdx-kernel/build.sh [hunk 3]
```diff
@@ -16,34 +16,34 @@ fi
...
-        tar xf ${PATCHSET}
+        git clone  -b ${UPSTREAM_TAG} --single-branch --depth 1 ${UPSTREAM_GIT_URI} "${SOURCE_DIR}"
+        tar xf "${PATCHSET}"
 
-        cd ${SOURCE_DIR}
-        git config user.name ${USER:-tdx-builder}
-        git config user.email ${USER:-tdx-builder}@$HOSTNAME
+        cd "${SOURCE_DIR}" || exit
+        git config user.name "${USER:-tdx-builder}"
+        git config user.email "${USER:-tdx-builder}"@"$HOSTNAME"
```
Comment (kenplusplus) [inline build/ubuntu-22.04/intel-mvp-tdx-kernel/build.sh:26]: please change ""@"" to "xxx @xxx"

File: build/ubuntu-22.04/intel-mvp-tdx-kernel/build.sh [hunk 4]
```diff
@@ -16,34 +16,34 @@ fi
...
         git submodule update --init
     fi
 }
 
 prepare() {
     echo "Prepare..."
-    cp ${CURR_DIR}/debian/ ${SOURCE_DIR} -fr
-    cp ${CURR_DIR}/debian.master/ ${SOURCE_DIR} -fr
-    cp ${CURR_DIR}/linux-5.15.0/* ${SOURCE_DIR} -fr
+    cp "${CURR_DIR}"/debian/ "${SOURCE_DIR}" -fr
```
Comment (kenplusplus) [inline build/ubuntu-22.04/intel-mvp-tdx-kernel/build.sh:36]: change to "${CURR_DIR}/debian/ ${SOURCE_DIR}"

**Review by kenplusplus — CHANGES_REQUESTED** [review]

**Review by ruomengh — COMMENTED** [review]
I would suggest to add a clear description on why the changes are needed.
For example: 
1. shell check scan issue *** 
2. fix method ***

File: build/ubuntu-22.04/intel-mvp-tdx-guest-grub2/build.sh (hunk 2 above)
Comment (tenderbulk) [reply build/ubuntu-22.04/intel-mvp-tdx-guest-grub2/build.sh:23]: shell check scan suggests adding "|| exit"

File: build/ubuntu-22.04/intel-mvp-tdx-kernel/build.sh (hunk 3 above)
Comment (tenderbulk) [reply build/ubuntu-22.04/intel-mvp-tdx-kernel/build.sh:26]: OK

File: build/ubuntu-22.04/intel-mvp-tdx-kernel/build.sh (hunk 4 above)
Comment (tenderbulk) [reply build/ubuntu-22.04/intel-mvp-tdx-kernel/build.sh:36]: OK

Comment (tenderbulk) [reply build/ubuntu-22.04/intel-mvp-tdx-kernel/build.sh:36]: seems this  copies directory : ${CURR_DIR}"/debian/    to    Dir : "${SOURCE_DIR}"

File: build/ubuntu-22.04/intel-mvp-tdx-guest-grub2/build.sh (hunk 2 above)
Comment (kenplusplus) [reply build/ubuntu-22.04/intel-mvp-tdx-guest-grub2/build.sh:23]: Ok, I found it is related to SC2164 https://www.shellcheck.net/wiki/SC2164

**Pull request closed without merging**